pkg/kubelet/kubelet.go

```diff
@@ -488,6 +495,41 @@ func NewMainKubelet(
 	return klet, nil
 }
 
+func getPodInfraContainerImagePullSecret(filename string) (*api.Secret, error) {
```

Comment:
nit: Why not read the secrets file contents in the `cmd/kubelet` package and inject it here, pkg/kubelet, as a secret?